src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -0,0 +1,644 @@
+    }
+    output_expr_ = prev_expr;
+  } else {
+    // Single table
+    if (node->GetList().size() == 1) node = node->GetList().at(0).Get();
+
+    // TODO(Ling): how should we determine the value of `is_for_update` field of logicalGet constructor?
+    output_expr_ = new OperatorExpression(
+        LogicalGet::Make(accessor_->GetDatabaseOid(node->GetDatabaseName()), accessor_->GetDefaultNamespace(),
+                         accessor_->GetTableOid(node->GetTableName()), {}, node->GetAlias(), false),
+        {});
+  }
+}
+
+void QueryToOperatorTransformer::Visit(parser::GroupByDescription *node, parser::ParseResult *parse_result) {}
+void QueryToOperatorTransformer::Visit(parser::OrderByDescription *node, parser::ParseResult *parse_result) {}
+void QueryToOperatorTransformer::Visit(parser::LimitDescription *node, parser::ParseResult *parse_result) {}
+void QueryToOperatorTransformer::Visit(UNUSED_ATTRIBUTE parser::CreateFunctionStatement *op,
+                                       parser::ParseResult *parse_result) {}
+
+void QueryToOperatorTransformer::Visit(UNUSED_ATTRIBUTE parser::CreateStatement *op,
+                                       UNUSED_ATTRIBUTE parser::ParseResult *parse_result) {}
+void QueryToOperatorTransformer::Visit(parser::InsertStatement *op, parser::ParseResult *parse_result) {
+  auto target_table = op->GetInsertionTable();
+  auto target_table_id = accessor_->GetTableOid(target_table->GetTableName());
+  auto target_db_id = accessor_->GetDatabaseOid(target_table->GetDatabaseName());
+  auto target_ns_id = accessor_->GetDefaultNamespace();
+
+  if (op->GetInsertType() == parser::InsertType::SELECT) {
+    auto insert_expr =
+        new OperatorExpression(LogicalInsertSelect::Make(target_db_id, target_ns_id, target_table_id), {});
+    op->GetSelect()->Accept(this, parse_result);
+    insert_expr->PushChild(output_expr_);
+    output_expr_ = insert_expr;
+    return;
+  }
+
+  // column_objects represents the columns for the current table as defined in its schema
+  auto column_objects = accessor_->GetSchema(target_table_id).GetColumns();
+
+  // vector of column oids
+  std::vector<catalog::col_oid_t> col_ids;
+
+  // INSERT INTO table_name VALUES (val1, val2, ...), (val_a, val_b, ...), ...
+  if (op->GetInsertColumns()->empty()) {
+    for (const auto &values : *(op->GetValues())) {
+      if (values.size() > column_objects.size()) {
+        throw CATALOG_EXCEPTION("ERROR:  INSERT has more expressions than target columns");
+      }
+      if (values.size() < column_objects.size()) {
+        for (auto i = values.size(); i != column_objects.size(); ++i) {
+          // check whether null values or default values can be used in the rest of the columns
+          if (!column_objects[i].Nullable() && column_objects[i].StoredExpression() == nullptr) {
+            throw CATALOG_EXCEPTION(
+                ("ERROR:  null value in column \"" + column_objects[i].Name() + "\" violates not-null constraint")
```

Comment:
Remove the `ERROR: ` prefix from this and all other exceptions. It's implied since we're throwing an exception.